File: tests/fixtures/sample.go
package main

import (
	"fmt"
	"strings"
)

// Reader defines a read interface
type Reader interface {
	Read(p []byte) (n int, err error)
}

// Writer defines a write interface
type Writer interface {
	Write(p []byte) (n int, err error)
}

// ReadWriter combines Reader and Writer
type ReadWriter interface {
	Reader
	Writer
}

// User represents a user in the system
type User struct {
	Name  string
	Email string
	Age   int
}

// Admin embeds User and adds Role
type Admin struct {
	User
	Role string
}

// NewUser creates a new User
func NewUser(name, email string, age int) *User {
	return &User{
		Name:  name,
		Email: email,
		Age:   age,
	}
}

// FullName returns the display name
func (u *User) FullName() string {
	return strings.Title(u.Name)
}

// Greet prints a greeting
func (u *User) Greet() {
	fmt.Printf("Hello, %s!\n", u.FullName())
}

// Promote creates an Admin from a User
func (u *User) Promote(role string) *Admin {
	return &Admin{User: *u, Role: role}
}

File: tests/fixtures/sample.rs
use std::fmt;
use std::collections::HashMap;

/// A trait for displayable items
pub trait Displayable {
    /// Format as string
    fn display(&self) -> String;
}

/// Represents a user
pub struct User {
    pub name: String,
    pub email: String,
    pub age: u32,
}

/// Status enum for users
pub enum Status {
    Active,
    Inactive,
    Banned(String),
}

impl User {
    /// Create a new user
    pub fn new(name: String, email: String, age: u32) -> Self {
        User { name, email, age }
    }

    /// Get user greeting
    pub fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }
}

impl Displayable for User {
    fn display(&self) -> String {
        format!("{} <{}>", self.name, self.email)
    }
}

/// Format a status value
pub fn format_status(status: &Status) -> String {
    match status {
        Status::Active => "active".to_string(),
        Status::Inactive => "inactive".to_string(),
        Status::Banned(reason) => format!("banned: {}", reason),
    }
}
